### PR TITLE
Add 'lerna exec' command

### PR DESCRIPTION
**Pull request by eirslett**
As opposed to 'lerna run', which runs
scripts defined in package.json, exec will
let you execute an inbuilt npm command directly
in each package, e.g.

- `lerna exec rebuild`
- `lerna exec publish`
- `lerna exec upgrade`

This is response to #127 and implements one of the solutions @thejameskyle suggests. What do you think?

### CODE REVIEW

**Comment by thejameskyle** [comment]
I meant `exec` as a way of running any command, not just npm commands.

For example:

```sh
$ lerna exec pwd
/path/to/repo/packages/package-1
/path/to/repo/packages/package-2
/path/to/repo/packages/package-3
```

**Comment by eirslett** [comment]
Ok, I updated the PR now! "lerna exec pwd" works as intended. It's still two commits, but they can easily be squashed on merge if that's better!

File: src/commands/ExecCommand.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  execute(callback) {
+    async.parallelLimit(this.packages.map(pkg => cb => {
+      this.runCommandInPackage(pkg, cb);
+    }), 4, callback);
+  }
+
+  runCommandInPackage(pkg, callback) {
+    ChildProcessUtilities.exec(this.command, { cwd: pkg.location }, (err, stdout) => {
```
Comment (thejameskyle) [inline src/commands/ExecCommand.js]: This should be handling more than just a single argument. It needs to handle things like:

```
lerna exec ls -la subdirectory 
```

It'd probably be better if you exposed the entire `process.argv` on `this.argv` inside commands and then trim the `lerna exec` from the beginning simply passing the rest through.

**Comment by thejameskyle** [comment]
Thinking more about this could you make it similar to how you pass flags to `npm run` commands in that you need `--` before it?

```sh
$ lerna exec -- ls -ls subdirectory
```

This will make it easier to add flags in the future:

```sh
$ lerna exec --debug -- ls -la subdirectory

File: src/commands/ExecCommand.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import Command from "../Command";
+import async from "async";
+
+export default class ExecCommand extends Command {
+  initialize(callback) {
+    this.command = this.input[0];
+    this.args = this.input.slice(1);
+
+    if (!this.command) {
+      callback(new Error("You must specify which npm command to run."));
```
Comment (thejameskyle) [inline src/commands/ExecCommand.js]: This error message is no longer valid

Comment (eirslett) [inline src/commands/ExecCommand.js]: fixed!

**Comment by joscha** [comment]
This is awesome, is there anything else that needs to be done to bring this in a mergeable state? Like the `--` functionality and/or a test with a command that has a few arguments?

**Comment by thejameskyle** [comment]
Yeah, I would like to see both of those things

**Comment by joscha** [comment]
@thejameskyle you can find the feature in #151 including a couple more tests and a fix for the test in this PR (it doesn't actually work properly).

**Comment by eirslett** [comment]
Thanks @joscha , I'll close this in favor of #151 .

**Pull request closed without merging**